conda_forge_tick/update_sources.py

```diff
@@ -354,7 +358,13 @@ def url_exists(url: str) -> bool:
     # For FTP servers an exception is not thrown
     if "No such file" in output.decode("utf-8"):
         return False
-    if "not retrieving" in output.decode("utf-8"):
+    if (
+        "not retrieving"
+        in output.decode(
+            "utf-8",
+        )
+        and "Remote file exists" not in output.decode("utf-8")
+    ):
```

Comment:
This is equivalent to not having this if condition at all. From my experience, `Remote file exists` doesn't mean anything if not retrieving is in the output.

Reply:
Well in my case the file exists but wget refuses to retrieve it:

```
(bot-dev) clarence:cf-scripts beckermr$ wget --spider https://eups.lsst.codes/stack/src/tags/w_2021_07.list
Spider mode enabled. Check if remote file exists.
--2021-02-12 12:04:56--  https://eups.lsst.codes/stack/src/tags/w_2021_07.list
Resolving eups.lsst.codes (eups.lsst.codes)... 104.155.133.93
Connecting to eups.lsst.codes (eups.lsst.codes)|104.155.133.93|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5414 (5.3K)
Remote file exists and could contain further links,
but recursion is disabled -- not retrieving.

(bot-dev) clarence:cf-scripts beckermr$ wget https://eups.lsst.codes/stack/src/tags/w_2021_07.list
--2021-02-12 12:05:03--  https://eups.lsst.codes/stack/src/tags/w_2021_07.list
Resolving eups.lsst.codes (eups.lsst.codes)... 104.155.133.93
Connecting to eups.lsst.codes (eups.lsst.codes)|104.155.133.93|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5414 (5.3K)
Saving to: ‘w_2021_07.list’

w_2021_07.list                             100%[======================================================================================>]   5.29K  --.-KB/s    in 0s      

2021-02-12 12:05:03 (10.6 MB/s) - ‘w_2021_07.list’ saved [5414/5414]

```

so your heuristic is wrong here.

Reply:
This message looks extremely common so I am inclined to merge this PR.

https://askubuntu.com/questions/389910/get-the-size-of-a-remote-file-from-terminal

Reply:
> so your heuristic is wrong here.

Giving one example where a heuristic is wrong and claiming that the heuristic is wrong does not make any sense to me.

Reply:
I don't want to debate semantics here. 

My point is simply that the code as written is failing when a url does in fact exist. Do you have an example url where wget returns "Remote file exists and could contain further links,
but recursion is disabled -- not retrieving." in the output and in fact the remote file does not exist?

Reply:
also this solves the github behavior:

```
In [3]: r = requests.head("https://github.com/beckermr/pizza-cutter/archive/0.2.5.tar.gz", allow_redirects=True)

In [4]: r
Out[4]: <Response [404]>

In [5]: r = requests.head("https://github.com/beckermr/pizza-cutter/archive/0.2.2.tar.gz", allow_redirects=True)

In [6]: r
Out[6]: <Response [200]>
```

so we should do that.